azurerm/resource_arm_data_lake_store.go

```diff
@@ -0,0 +1,193 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/datalake/store/mgmt/2016-11-01/account"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataLakeStore() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDateLakeStoreCreate,
+		Read:   resourceArmDateLakeStoreRead,
+		Update: resourceArmDateLakeStoreUpdate,
+		Delete: resourceArmDateLakeStoreDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
```

Comment:
Could we get some validation on this property? the docs say 3-24 so at the very lease we should check that. Ideally we should use regex to also validate the characters.

Reply:
done in next commit